handlers/vm/base.go

```diff
@@ -12,6 +23,84 @@ var SupportedAPICalls = []string{
 	"/api.Instance/Log",
 }
 
+func (*Base) ValidateAuthenticationType(in json.RawMessage) (json.RawMessage, model.AuthenticationType, error) {
+	var json_template struct {
+		AuthenticationType string `json:"authentication_type,omitempty"`
+	}
+
+	if err := json.Unmarshal(in, &json_template); err != nil {
+		return nil, 0, err
+	}
+	var ret model.AuthenticationType
+	if json_template.AuthenticationType != "" {
+		format, ok := model.AuthenticationType_value[strings.ToUpper(json_template.AuthenticationType)]
+		if !ok {
+			return nil, 0, errors.Errorf("Unknown value at format: %s", json_template.AuthenticationType)
+		}
+		ret = model.AuthenticationType(format)
+
+		// Remove authentication_type field
+		tmp := make(map[string]interface{})
+		var err error
+		if err = json.Unmarshal(in, &tmp); err != nil {
+			return nil, 0, errors.Wrap(err, "Failed json.Unmarshal")
+		}
+		delete(tmp, "authentication_type")
+		in, err = json.Marshal(tmp)
+		if err != nil {
+			return nil, 0, errors.Wrap(err, "Failed json.Marshal")
+		}
+	}
+	return in, ret, nil
+}
+
+func (*Base) ValidatePublicKey(h handlers.ResourceHandler, authType model.AuthenticationType, sshPubKey string) error {
+	switch authType {
+	case model.AuthenticationType_NONE:
+	case model.AuthenticationType_PUB_KEY:
+		if sshPubKey == "" {
+			return handlers.ErrInvalidTemplate(h, "ssh_public_key is not set")
+		}
+
+		isValidate := validatePublicKey([]byte(sshPubKey))
```

Comment:
Just call ``ssh.ParsePublicKey()`` then return the error from the function. This function squash the error details from ``ssh.ParsePublicKey()``.